sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1034,7 +1038,7 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 				.dailink = {SDW_AMP_OUT_DAI_ID, SDW_UNUSED_DAI_ID},
 				.init = sof_sdw_cs42l43_spk_init,
 				.rtd_init = cs42l43_spk_rtd_init,
-				.quirk = SOF_CODEC_SPKR,
+				.quirk = SOF_CODEC_SPKR | SOF_SIDECAR_AMPS,
```

Comment:
Oh, I am so lost here.

How does one make the difference where cs42l43 is used with or without the sidecar amps?

Or is this a list of supported quirks, and the actual selection is done with e.g. a DMI quirk?

Reply:
These are not adding the quirk but marking this DAI link as only being included if the quirk is present, the presence of the quirk is still specified through sof_sdw_quirk:

https://github.com/thesofproject/linux/pull/4848/commits/47bae326da237ca6cf0414dc5a8a823d1ff83a39

The idea is that we don't add the speaker DAI link unless either the SOF_CODEC_SPKR or the SOF_SIDECAR_AMPS quirks are present, as those are the two things that would use the cs42l43 speaker DAI.

Reply:
oh, ok, so that's this the same case as before, for an actual SKU using this hardware layout we'd need a DMI quirk, yes?

Reply:
Correct yeah would need a DMI quirk.